cms/models.py

```diff
@@ -40,17 +72,77 @@ class HomePage(Page):
         help_text="The subtitle text to display in the hero section of the home page.",
     )
 
+    product_section_title = models.CharField(
+        max_length=255,
+        null=True,
+        blank=True,
+        help_text="The tile text to display in the product cards section of the home page.",
+    )
+
     content_panels = Page.content_panels + [
         ImageChooserPanel("hero"),
         FieldPanel("hero_title"),
         FieldPanel("hero_subtitle"),
+        FieldPanel("product_section_title"),
+        InlinePanel("featured_products", label="Featured Products"),
     ]
+
     parent_page_types = [Page]
     subpage_types = [
-        "CoursePage",
         "ResourcePage",
     ]
 
+    def _get_child_page_of_type(self, cls):
+        """Gets the first child page of the given type if it exists"""
+        child = self.get_children().type(cls).live().first()
+        return child.specific if child else None
+
+    @property
+    def product_cards(self):
+        pages = []
+        for page in self.featured_products.all():
+            if page.course_product_page:
+                pages.append(page.course_product_page.specific)
+        return pages
+
+    def get_context(self, request, *args, **kwargs):
+        return {
+            **super().get_context(request),
+            "product_cards_section_title": self.product_section_title,
+            "product_cards": self.product_cards,
+        }
```

Comment:
Like I mentioned in the product detail page PR, it's strongly preferred to do calculation and deep variable access in the model and pass it down to the template whenever we can. Suggestions here also have some small variable name changes:

```python
    @property
    def products(self):
        page_data = []
        for page in self.featured_products.all():
            if page.course_product_page:
                product_page = page.course_product_page.specific
                page_data.append({
                    "title": product_page.title,
                    "description": product_page.description,
                    "feature_image": product_page.feature_image,
                    "start_date": product_page.product.first_unexpired_run
                })
        return page_data

    def get_context(self, request, *args, **kwargs):
        return {
            **super().get_context(request),
            "product_section_title": self.product_section_title,
            "products": self.products,
        }
```